### ISSUE TITLE
Simplify component construction

### ISSUE THREAD

**Issue by jonnenauha** [realXtend/tundra#754]
Recently there was some changes that made it possible to create components with a null scene to the constructor. I think before it we didn't have this problem and they were always created to a owning scene. This was a good change, we no longer need a dummy scene to put stuff into for inspection etc. Afaik this feature is mostly used when you drag and drop a .txml into Tundra and want a UI preview without actually adding it to the active scene yet. (This is what I know happens, not sure if it was the reasoning to allow creating components this way).

Anyhow, now we have a slight problem that has been going on for a while. Components that are not being maintained that well or by programmer error are crashing on this kind of creation. The feature is rarely used so it goes unnoticed untill someone does a .txml import like this.

If you use the ctor given `Scene*` or `Framework*` in the ctor youll get a crash like this. We have pretty much fixed all current components (just fixed couple ones in core and meshmoon today), but this makes writing components hard. Having to know to check for null ptrs.

Also sometimes choosing the logic to be done in ctor or handing it during component lifetime and on the `ParentEntitySet` signal is tricky. It is not well known to even me if `ParentEntitySet` signal is emitted **before** `AttributeChanged()` is invoked for the first time with the initial attribute values. So lets say that you want to make a `AssetRefListener` that gets used in `AttributeChanged`, but you only want to do that allocation if you are on a non-headless Tundra instance. As the framework cannot be reliably used in the ctor, you move the headless check and allocation into a slot connected to `ParentEntitySet` signal. Now, is your ref listener still null when you are handling the first value of your asset ref attribute in `AttributeChanged()`? Or maybe I want to do certain allocations only on a server instance?

This kind of confusion makes it harder to write components. If I cant figure out the order quickly, I would think more novice Tundra users will have immense problems following the initialization logic/order of a component.

Hence I propose that all component ctors should get a valid `Framework*` into them, that should be passed to the `IComponent` ctor and be set to the `framework` (currently set from the scene ptr, and left null if that is null) member variable. This way only thing that you need to be aware is if the scene is null. This change should not break anything as nothing but the component factories "new" up components. Framework could be passed to the factories to be used on component creation.

We should also document if the `ParentEntitySet` signal is guaranteed to be emitted before `AttributesChanged()` is invoked.

(Another option suggested by @Stinkfist0 would be to have `Initialize()` function you can override that will get called once when everything is in order.)

**TL;DR** We need to change so that framework ptr is always valid in component constructors.
